crm/crm/doctype/opportunity/opportunity.js

```diff
@@ -300,6 +300,22 @@ crm.Opportunity = class Opportunity extends frappe.ui.form.Controller {
 			party: this.frm.doc.party_name,
 		});
 	}
+
+	contact_schedule_add() {
+		const table = this.frm.doc.contact_schedule;
+		const last_row = table?.[table.length - 1];
+
+		if (!last_row) return;
+
+		this.auto_fill_sales_person(last_row);
```

Comment:
do not assume added row will be last, rely on arguments to get the added row reference